--- a/src/moepkg/ui.nim
+++ b/src/moepkg/ui.nim
@@ -182,30 +182,4 @@
-let libncurseswHandle: LibHandle =
-  # Workaround for `could not import: init_extended_color`.
-  # The LibHandle for checking availability of "init_extended_color" and "init_extended_pair" when running.
-  # https://github.com/fox0430/moe/issues/2262
-  # https://github.com/fox0430/moe/issues/2146
-  block:
-    when defined(windows):
-      const libPatterns = ["libncurses.dll"]
-    elif defined(macosx):
-      const libPatterns = ["libncurses.dylib"]
-    else:
-      const libPatterns = ["libncursesw.so.6", "libncursesw.so.5", "libncursesw.so"]
-
-    var lib: LibHandle
-    for libncurses in libPatterns:
-      lib = loadLib(libncurses)
-      if not lib.isNil:
-        break
-
-    if lib.isNil:
-      echo "Error: libncurses not found. Please install ncurses"
-      quit()
-
-    lib
-
-let
-  ncursesInitExtendedColor =
-    cast[NcursesInitExtendedColor](libncurseswHandle.symAddr("init_extended_color"))
-  ncursesInitExtendedPair =
-    cast[NcursesInitExtendedPair](libncurseswHandle.symAddr("init_extended_pair"))
+proc ncursesInitExtendedColor(color: cint, r, g, b: cint): ErrCode
+  {.cdecl, importc: "init_extended_color".}
+proc ncursesInitExtendedPair(pair, f, b: cint): ErrCode
+  {.cdecl, importc: "init_extended_pair".}
@@ -217,1 +191,1 @@
-  not ncursesInitExtendedColor.isNil and not ncursesInitExtendedPair.isNil
+  true
